src/overlay/PeerDoor.cpp

```diff
@@ -31,6 +32,7 @@ PeerDoor::start()
         CLOG(DEBUG, "Overlay") << "PeerDoor binding to endpoint " << endpoint;
         mAcceptor.open(endpoint.protocol());
         mAcceptor.set_option(asio::ip::tcp::acceptor::reuse_address(true));
+        fd::disableProcessInheritance(mAcceptor);
```

Comment:
note that there is a very small race condition here: if fork is called from a different thread between the call to "open" and this call, it will be too late.
This is why `SOCK_CLOEXEC` can be passed in to `socket` ; problem it seems is that I don't think there is a good way to fix this with the way asio works (let me know if you find otherwise)

Reply:
Yeah, I'm aware of the race condition but I'm not sure how well that plays with asio. This is mostly an issue in tests right? And I don't know how the code flow works but is the `PeerDoor` created in a non multi threaded scenario?

Anyway, I'll try to see if there's a way to do this properly on asio. There's the issue of accepted sockets as well, for which this should use that same flag on `accept` but there doesn't seem to be an easy way to do this with asio.

Reply:
It's not just in tests: we have worker threads (look for usage of `getWorkerIOService()`) that run in parallel. Right now we do not fork processes from those, but I imagine we will as it's the kind of work that we can easily punt to the side. If you can't find a simple solution that's fine for now: what we may be able to do in the future is do "cooperative forking" and not call fork when we know that there is a possible chance of hitting this race condition

Reply:
I see, that makes sense. This is such stupid problem (and funnily enough we hit the same issue at work last week) but there doesn't seem to be an easy way to mitigate it. I found [this glib thread](https://mail.gnome.org/archives/gtk-devel-list/2015-March/msg00038.html) which is interesting but probably not the [best way](https://mail.gnome.org/archives/gtk-devel-list/2015-March/msg00044.html) to solve it. I'll have another look sometime this week.